doc/architectural/restrict-function-pointer.md

```diff
@@ -0,0 +1,182 @@
+
+\page restricting-function-pointers Restricting function pointers
```

Comment:
As above: I'm ok with restoring it, but I do note that the man page has this content now.

Reply:
Removed (and links from user guide and architectural guide).